src/server/node_manager.c

```diff
@@ -8776,3 +8779,57 @@ update_node_rassn(attribute *pexech, enum batch_op op)
 {
 	update_job_node_rassn(NULL, pexech, op);
 }
+
+static void
+set_resv_for_degrade(struct pbsnode *pnode, resc_resv *presv)
+{
+	char *str_time;
+	long degraded_time;
+
+	if (presv->ri_wattr[RESV_ATR_resv_standing].at_val.at_long == 0)
+		presv->ri_degraded_time = presv->ri_wattr[RESV_ATR_start].at_val.at_long;
+	else
+		find_degraded_occurrence(presv, pnode, Set_Degraded_Time);
+
+	degraded_time = presv->ri_degraded_time;
+
+	if ((degraded_time > (time_now + reserve_retry_cutoff)) &&
+		(degraded_time > (time_now + reserve_retry_init)))
+			set_resv_retry(presv, (time_now + reserve_retry_init));
+
+	if (presv->ri_resv_retry) {
+		str_time = ctime(&presv->ri_resv_retry);
+		if (str_time != NULL) {
+			str_time[strlen(str_time) - 1] = '\0';
+			(void) snprintf(log_buffer, sizeof(log_buffer),
+				"An attempt to reconfirm reservation will be made on %s",
+				str_time);
+			log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_RESV, LOG_NOTICE,
+				presv->ri_qs.ri_resvID, log_buffer);
+		}
+	}
+	(void) resv_setResvState(presv, presv->ri_qs.ri_state, RESV_DEGRADED);
+
+	/* the number of vnodes down could exceed the number of vnodes in
+	 * the reservation only in the case of a standing reservation for
+	 * which the vnodes unavailable are associated to later occurrences
+	 */
+	if (presv->ri_vnodes_down > presv->ri_vnodect) {
+		attribute *rsv_attr = presv->ri_wattr;
+		/* If a standing reservation we print the execvnodes sequence
+		 * string for debugging purposes */
+		if (rsv_attr[RESV_ATR_resv_standing].at_val.at_long) {
+			snprintf(log_buffer, sizeof(log_buffer), " execvnodes sequence %s",
+				rsv_attr[RESV_ATR_resv_execvnodes].at_val.at_str);
+			log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_RESV, LOG_DEBUG,
+				presv->ri_qs.ri_resvID, log_buffer);
```

Comment:
DEBUG is printed at the default logging level.  This seems to me like a lengthy debug message that should be at a higher log level.